radix-engine/src/track/track.rs

```diff
@@ -787,17 +809,20 @@ impl<'s, S: SubstateDatabase, M: DatabaseKeyMapper> SubstateStore for Track<'s,
         }
 
         // Optimization, no need to go into database if the node is just created
-        if is_new {
-            return (items, store_access);
+        if items.len() == count || is_new {
```

Comment:
Can we remind me the expect behaviour regarding whether it should be exact or limit?

Reply:
Also, can we rename `count` to `limit` everywhere?